src/algebra/ring/ulift.lean

```diff
@@ -82,4 +82,17 @@ by refine_struct { zero := (0 : ulift α), one := 1, add := (+), mul := (*), sub
   zsmul := sub_neg_monoid.zsmul };
 tactic.pi_instance_derive_field
 
+instance nontrivial [nontrivial α] : nontrivial (ulift α) :=
+let ⟨x, y, h⟩ := @nontrivial.exists_pair_ne α _ in
```

Comment:
Don't we have `function.injective.nontrivial`?

Reply:
nice, thanks! seems sadly that the `..` syntax doesn't work in `refine_struct`, though; it pretends to work but actually just creates an mvar